### ISSUE TITLE
visit stuff about current post

### ISSUE THREAD

**Issue by nexushoratio** [nexushoratio/userscripts#23]
Somewhere on a given post, there is a way to click on it and open it up by itself.  Sometimes I click it by accident, but I can never do it on purpose.

I can force the issue with meatball menu -> copy link to post -> view post.  And that goes to /posts/....

Also, one can more easily click on the original poster (person or company), the influencer, the group, etc.

While a post has focus, it should be possible to do all of those with keys.  Possible `v` for visit (or view).

- `v p` view post
- `v g` view group
- `v r` view reactions
- `v P` view Poster
- `v i` view influencer   (this could get tricky as a liked repost will have OP, reposter, influencer)

**Comment by nexushoratio** [comment]
And to be recursive, most of these should work when viewing the standalone post as well.

(Same goes for comments actually).

I wonder if it would make sense to have `v c` instead of `c` for comments then.

**Comment by nexushoratio** [comment]
Hmmm.  `v p` isn't that easy after all.  Very few posts actually have a self referential link.  I can only find one in my feed so far.

While it would be easy to generate the link from the `data-urn` field on the post itself, I'm not sure what will happen if we try to navigate to it directly.  Will it break the SPA aspect of LI?

That's one reason we've been keen on the `button.click()` approach.

**Comment by nexushoratio** [comment]
Just a guess, but it looks like a reaction[^1] on a repost[^2] is what results in a link-to-details-page.

Actually, the reaction is not necessary.  Anything that causes it to be in your feed, including a primary connection that makes the repost.

[^1]: A reaction can be a human like, but also appears to be a generated thing like `New activity in this group`
[^2]: A repost is one case, but it looks like raising a comment to a post is equivalent.

**Comment by nexushoratio** [comment]
I've not tried, but I suspect that `Repost with your thoughts` is the same thing as commenting and starting a post.

I wonder how deep it could go.

**Comment by nexushoratio** [comment]
Oh, and that means the (⋯) menu -> Copy link to post goes to the repost, and the link-to-details-page goes to the original post.

Since the `v p` sequence is intended to emulate the Copy link to post experience, we might have to get creative.  Like creating a link and clicking on it.  I don't feel bad about modifying the html to add the keyboard help screen.  But, modifying a post might be a tad invasive.

The toast the pops up after copying a link, with the `View post` link it in... well, that link is a pretty bare href with no obvious event handlers on it.  So maybe clicks are being caught at the PSA level and this wouldn't cause a page reload.

**Comment by nexushoratio** [comment]
I was wrong.  The `View post` link DOES cause a reload.

**Comment by nexushoratio** [comment]
Or sometimes the `link-to-details` link does NOT cause a reload.  So confused.

**Comment by nexushoratio** [comment]
Current steps.

I can reproduce the experience with a single repost view with VM turned off.

Currently, I'm focusing on the anchor element, and remove stuff from inside of it (divs, spans, etc).  And see when it still works and when it doesn't.  What is the minimal needed.

**Comment by nexushoratio** [comment]
In FF Dev Tools inspector.

Right click on anchor and `Use in console` (and get a tempN variable).
```
tempN.innerHTML=null
tempN.click()
```
Does SPA style navigation.  Yay.

Go back (and have to do find the node again because it was rebuilt from scratch by refetching the data), set innerHTML to null again.  This time duplicate node.  Rename either node.

The old one will still click SPA style, new one will not.  Even rearranging doesn't seem to matter.

Cannot yet figure out what the difference is.

**Comment by nexushoratio** [comment]
`for (a in temp0) {console.debug(temp0[a], temp1[a])}`

No obvious differences.

**Comment by nexushoratio** [comment]
Since I added `f`ocus support, I don't think there is much need to support all of the various things suggested in the original report.

`f` then `TAB` `TAB` `TAB` actually works pretty well, since most of the items of interest are near the top.  And since posts are different, having keys that work on some and not others would be pretty frustrating, as I discovered while implementing viewing reactions.
